service/src/test/java/bio/terra/tanagra/service/accesscontrol/AouWorkbenchAccessControlTest.java

```diff
@@ -39,31 +41,14 @@ protected void cleanup() {
   @Test
   void underlay() {
     // isAuthorized, getPermissions, listAllPermissions, listAuthorizedResources
-    ResourceId cmsSynpufId = ResourceId.forUnderlay(CMS_SYNPUF);
     ResourceId aouSyntheticId = ResourceId.forUnderlay(AOU_SYNTHETIC);
-    ResourceId sddId = ResourceId.forUnderlay(SDD);
-    assertHasPermissions(USER_1, cmsSynpufId);
-    assertHasPermissions(USER_1, aouSyntheticId);
-    assertHasPermissions(USER_1, sddId);
-    assertHasPermissions(USER_2, cmsSynpufId);
-    assertHasPermissions(USER_2, aouSyntheticId);
-    assertHasPermissions(USER_2, sddId);
-    assertHasPermissions(USER_3, cmsSynpufId);
-    assertHasPermissions(USER_3, aouSyntheticId);
-    assertHasPermissions(USER_3, sddId);
-    assertHasPermissions(USER_4, cmsSynpufId);
-    assertHasPermissions(USER_4, aouSyntheticId);
-    assertHasPermissions(USER_4, sddId);
-
+    assertUnderlayPermissions(USER_1, aouSyntheticId);
```

Comment:
Is it possible to reuse an existing method here instead? e.g. `assertDoesNotHavePermissions(USER_1, aouSyntheticId)`

Reply:
No it does not work since we return `true` for `isAuthorized` call. The `assertHasPermissions(USER_1,aouSyntheticId)` does not work because the assertion methond assumes that if isAuthorized is true then listAllPermissions must also asert to true see [assertHasPermissions](https://github.com/DataBiosphere/tanagra/blob/90ed11eb51a6413b512367a0c6a3af8c27ca7801/service/src/test/java/bio/terra/tanagra/service/accesscontrol/BaseAccessControlTest.java#L198-L200).  This is the weirdness for underlay

Reply:
Ah I see -- `isAuthorized` always returns true for a particular underlay, but `listAuthorizedResources` always returns empty when asked to list the underlays. That is confusing. Could you add a comment here to that effect please, so the next person doesn't get tripped up on the same thing?

Ditto for studies, since I think that will be similar (`isAuthorized` sometimes returns true for a particular study, but `listAuthorizedResources` always returns empty when asked to list the studies).